### DISCUSSION TITLE
Restrict REST or GraphQL API to certain domains

### DISCUSSION THREAD

**Discussion by dominikfoeger** [statamic/cms#4354] [Q&A] [answered]
Hi there,

did anyone manage to restrict the REST or GraphQL endpoints to certain domains? I want my decoupled frontend to be able to access it, but no one else.

I tried using the Laravel CORS config (Fruitcake middleware) but I'm still able to query all entries (inclunding unpublished entries) via local postman. Maybe I'm misunderstanding CORS or I need to write a custom middleware…

Thanks

**Comment by edalzell** [comment, marked as answer]
ya middleware is the answer here, if you don't want to PR into core.

**Reply by dominikfoeger** [reply]
Thanks, Erin!
